dLux/apertures.py

```diff
@@ -0,0 +1,2641 @@
+        rotation: float, radians
+            The rotation of the aperture away from the positive 
+            x-axis. 
+        """
+        super().__init__(
+            centre = centre, 
+            strain = strain, 
+            compression = compression,
+            rotation = rotation,
+            occulting = occulting,
+            softening = softening)
+        self.nsides = int(nsides)
+        self.rmax = np.array(rmax).astype(float)
+        
+        
+    def _extent(self: ApertureLayer) -> float:
+        """
+        Returns the largest distance to the outer edge of the aperture from the
+        centre. For inherited classes, consider implementing analytically for speed.
+
+        Returns
+        -------
+        extent : float
+            The maximum distance from centre to edge of aperture
+        """
+        return self.rmax
+        
+    
+    def _metric(self: ApertureLayer, coords: float) -> float:
+        """
+        A measure of how far a pixel is from the aperture.
+        This is a very abstract description that was constructed 
+        when dealing with the soft edging. For a normal binary 
+        representation the metric is zero if it is inside the
+        aperture and one if it is outside the aperture. Notice,
+        we have not attempted to prove that this is a metric 
+        via the axioms, this is just a handy name that brings 
+        to mind the general idea. For a soft edged aperture the 
+        metric is different.
+
+        Parameters:
+        -----------
+        distances: Array
+            The distances of each pixel from the edge of the aperture. 
+            Again, the words distances is designed to aid in 
+            conveying the idea and is not strictly true. We are
+            permitting negative distances when inside the aperture
+            because this was simplest to implement. 
+
+        Returns:
+        --------
+        non_occ_ap: Array 
+            This is essential the final step in processing to produce
+            the aperture. What is returned is the non-occulting 
+            version of the aperture. 
+        """
+        x: float = coords[0]
+        y: float = coords[1]
+
+        neg_pi_to_pi_phi: float = np.arctan2(y, x) 
+        alpha: float = np.pi / self.nsides
+            
+        i: int = np.arange(self.nsides)[:, None, None] # Dummy index
+        bounds: float = 2. * i * alpha
+        phi: float = self._offset(neg_pi_to_pi_phi, bounds[0])
+            
+        wedges: float = self._make_wedges(phi, bounds)
+        ms: float = -1 / np.tan(2. * i * alpha + alpha)
+        xs: float = self.rmax * np.cos(2. * i * alpha)
+        ys: float = self.rmax * np.sin(2. * i * alpha)
+        dists: float = self._perp_dists_from_lines(ms, xs, ys, x, y)
+        inside: float = self._is_orig_left_of_edge(ms, xs, ys)
+         
+        dist: float = (inside * dists * wedges)
+        return self._soften(dist.sum(axis=0))
```

Comment:
This function uses the different syntax of val : dtype = ... - I would remove these just to keep consistency across the package